optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/main/java/org/optaplanner/quarkus/deployment/DotNames.java

```diff
@@ -27,11 +42,48 @@
 
     static final DotName PLANNING_SOLUTION = DotName.createSimple(PlanningSolution.class.getName());
     static final DotName PLANNING_ENTITY = DotName.createSimple(PlanningEntity.class.getName());
+
+    static final DotName CONSTRAINT_CONFIGURATION_PROVIDER =
+            DotName.createSimple(ConstraintConfigurationProvider.class.getName());
+    static final DotName CONSTRAINT_WEIGHT = DotName.createSimple(ConstraintWeight.class.getName());
+    static final DotName PLANNING_PIN = DotName.createSimple(PlanningPin.class.getName());
+    static final DotName PLANNING_ID = DotName.createSimple(PlanningId.class.getName());
+    static final DotName PLANNING_ENTITY_COLLECTION_PROPERTY =
+            DotName.createSimple(PlanningEntityCollectionProperty.class.getName());
+    static final DotName PLANNING_ENTITY_PROPERTY = DotName.createSimple(PlanningEntityProperty.class.getName());
+    static final DotName PLANNING_SCORE = DotName.createSimple(PlanningScore.class.getName());
+    static final DotName PROBLEM_FACT_COLLECTION_PROPERTY = DotName.createSimple(ProblemFactCollectionProperty.class.getName());
+    static final DotName PROBLEM_FACT_PROPERTY = DotName.createSimple(ProblemFactProperty.class.getName());
+    static final DotName VALUE_RANGE_PROVIDER = DotName.createSimple(ValueRangeProvider.class.getName());
+    static final DotName ANCHOR_SHADOW_VARIABLE = DotName.createSimple(AnchorShadowVariable.class.getName());
+    static final DotName CUSTOM_SHADOW_VARIABLE = DotName.createSimple(CustomShadowVariable.class.getName());
+    static final DotName INVERSE_RELATION_SHADOW_VARIABLE = DotName.createSimple(InverseRelationShadowVariable.class.getName());
+    static final DotName PLANNING_VARIABLE = DotName.createSimple(PlanningVariable.class.getName());
+    static final DotName PLANNING_VARIABLE_REFERENCE = DotName.createSimple(PlanningVariableReference.class.getName());
```

Comment:
Order: PLANNING_VARIABLE before any the shadow vars types.
Typical order is something like this: solution related stuff, entity related stuff, genuine planning variable related stuff (IIRC including value range providers), shadow vars related.
BAsically, when the annotations are parsed by the Descriptors, their order is the order in which they get parsed, mostly.

Motivation for why order matter: if code always handles a set of aspects in the same order, it is much, much clearer when something is missing, in which order they get filled in (important to avoid chicken-and-egg situations if they depend on each other) and sometimes even avoids a deadlock.